### PR TITLE
Fixed reader to support both authentication and tenant-id

### LINKED ISSUE #5718: Loki-canary can't use both authentication and x-scope-orgid (tenant-id)
**Describe the bug**
Loki-canary is unable to use both authentication and tenant-id settings at the same time.

**To Reproduce**
1. Set username and password with the -user and -pass arguments
2. Set tenant-id with the -tenant-id argument

**Expected behavior**
Users should be able to set both headers at the same time.

**Environment:**
 - Infrastructure: all
 - Deployment tool: all

**Screenshots, Promtail config, or terminal output**
https://github.com/grafana/loki/blob/6d7422aabf7fa41c7ef527958cb4fda1566384b4/pkg/canary/reader/reader.go#L88-L92

### PR DESCRIPTION
**Pull request by kovaxur**
**What this PR does / why we need it**:
Loki-canary will be able to use both basic-auth and x-scope-orgid headers.

**Which issue(s) this PR fixes**:
Fixes #5718 

**Special notes for your reviewer**:

**Checklist**
- [X] Documentation added
- [N/A] Tests updated
- [X] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by sandeepsukhani** [comment]
It also won't work as is because the header setting you have changed is used just for live tailing. You also need to update `QueryCountOverTime` and `Query` functions below.

File: docs/sources/operations/loki-canary.md [hunk 1]
```diff
@@ -355,6 +355,8 @@ All options:
   -streamvalue string
         The unique stream value for this instance of Loki Canary to use in the log selector
         (default "stdout")
+  -tenant-id string
+        Tenant id to be set in X-Scope-OrgID header.
```
Comment (KMiller-Grafana) [inline docs/sources/operations/loki-canary.md:359]:
```suggestion
        Tenant ID to be set in X-Scope-OrgID header.
```

File: cmd/loki-canary/main.go [hunk 2]
```diff
@@ -37,9 +37,9 @@ func main() {
 	port := flag.Int("port", 3500, "Port which loki-canary should expose metrics")
 	addr := flag.String("addr", "", "The Loki server URL:Port, e.g. loki:3100")
 	tls := flag.Bool("tls", false, "Does the loki connection use TLS?")
-	user := flag.String("user", "", "Loki username. Must not be set with tenant-id flag.")
-	pass := flag.String("pass", "", "Loki password. Must not be set with tenant-id flag.")
-	tenantID := flag.String("tenant-id", "", "Tenant id to be set in X-Scope-OrgID header. Must not be set with user/pass flags.")
+	user := flag.String("user", "", "Loki username.")
+	pass := flag.String("pass", "", "Loki password.")
+	tenantID := flag.String("tenant-id", "", "Tenant id to be set in X-Scope-OrgID header.")
```
Comment (KMiller-Grafana) [inline cmd/loki-canary/main.go:42]:
```suggestion
	tenantID := flag.String("tenant-id", "", "Tenant ID to be set in X-Scope-OrgID header.")
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
2 tiny doc changes.

**Review by KMiller-Grafana — APPROVED** [review]
From a Docs perspective, I approve this PR.

**Review by sandeepsukhani — APPROVED** [review]
LGTM

**Pull request merged**